lib/internal/test_runner/utils.js

```diff
@@ -74,10 +86,71 @@ function convertStringToRegExp(str, name) {
   }
 }
 
+const kBuiltinDestinations = new SafeMap([
+  ['stdout', process.stdout],
+  ['stderr', process.stderr],
+]);
+
+const kBuiltinReporters = new SafeMap([
+  ['spec', 'node:test/reporter/spec'],
+  ['dot', 'node:test/reporter/dot'],
+  ['tap', 'node:test/reporter/tap'],
+]);
+
+const kDefaultReporter = 'tap';
+const kDefaltDestination = 'stdout';
+
+async function getReportersMap(reporters, destinations) {
+  return SafePromiseAllReturnArrayLike(reporters, async (name, i) => {
+    const destination = kBuiltinDestinations.get(destinations[i]) ?? createWriteStream(destinations[i]);
+    let reporter = await requireOrImport(kBuiltinReporters.get(name) ?? name);
+
+    if (reporter && reporter.default) {
+      reporter = reporter.default;
+    }
+
+    if (reporter.prototype && ObjectGetOwnPropertyDescriptor(reporter.prototype, 'constructor')) {
```

Comment:
If we are guarding above, we should probably guard here to, or remove the guarding above.
```suggestion
    if (reporter?.prototype && ObjectGetOwnPropertyDescriptor(reporter.prototype, 'constructor')) {
```